dataset/dataset.go

```diff
@@ -446,8 +449,14 @@ func (c *Client) GetEditions(ctx context.Context, userAuthToken, serviceAuthToke
 }
 
 // GetVersions gets all versions for an edition from the dataset api
-func (c *Client) GetVersions(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceAuthToken, collectionID, datasetID, edition string) (m []Version, err error) {
+func (c *Client) GetVersions(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceAuthToken, collectionID, datasetID, edition string, q *QueryParams) (m VersionsList, err error) {
 	uri := fmt.Sprintf("%s/datasets/%s/editions/%s/versions", c.hcCli.URL, datasetID, edition)
+	if q != nil {
+		if err := q.Validate(); err != nil {
+			return VersionsList{}, err
```

Comment:
This method uses named return vars so for consistency and to avoid confusion you should return the named vars. A function that declares named return vars implies the func will return those specific vars so it's misleading and potentially confusing to return other vars. I'm aware this is picky but I think its an important point to adhere to when using a this feature.

Update  `if err := q.Validate...`  to  `if err = q.Validate...`  i.e. Assign the err returned from the function to the existing ` err` named parameter.

Then simply 
```
return
```

Reply:
👍